### PR TITLE
Panel: use align-items for cross axis alignment.

### LINKED ISSUE #12020: Panel Body CSS Shift IE11
### Environment Information

- **Package version(s)**: at least 7.86.0 and forward
- **Browser and OS versions**: IE11

### Please provide a reproduction of the bug in a codepen:
https://developer.microsoft.com/en-us/fabric#/controls/web/panel

Open any of the first two panels in IE11, you can see they are rendered completely unusable.
image

image

#### Actual behavior:

Inner content shifted outside of the panel
Unclear if this is the root cause, but by switching the css property in the class ms-Panel-header, margin-right: auto to margin-right:0px seems to at least display the content

#### Expected behavior:

All content displayed appropriately

### Priorities and help requested:

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: Blocking / High

Products/sites affected: All Panel using sites, including SharePoint/MSIT

### PR DESCRIPTION
**Pull request by aneeshack4**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #12020 
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Panel's header styles are broken in IE11 because IE11 isn't compliant with the W3 spec on margins - instead of changing the marginRight auto setting (which would break the other browsers), I'm fixing the other problem which is that with flexbox, vertical alignment (cross axis) should be set with alignItems, not justifyContent (that's for the main axis). 

#### Focus areas to test
IE 11:
image
Edge:
image
Chrome:
image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12025)

### CODE REVIEW

**Comment by aneeshack4** [comment]
@betrue-final-final could you look over the screener changes? https://screener.io/v2/states/ryQqxfIYZ.OfficeDev-office-ui-fabric-react/panel_header

File: packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts [hunk 1]
```diff
@@ -286,7 +286,7 @@ export const getStyles = (props: IPanelStyleProps): IPanelStyles => {
       classNames.navigation,
       {
         display: 'flex',
-        justifyContent: 'flex-end'
+        alignItems: 'flex-end'
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: wait, what's the intention here? why is this needed?
isn't there a margin property that needs to get removed?

Comment (phkuo) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: we likely want justify-content:space-between to force the title and X button to be on the edges

Comment (maxwellred) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: I just tried out space-between successfully

Comment (aneeshack4) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: same tested it on IE 11, chrome, and edge again all work

**Comment by phkuo** [comment]
@aneeshack4 please also clean up any extraneous margins. Flexed children should avoid using margins altogether, as a general guideline.
https://github.com/OfficeDev/office-ui-fabric-react/blob/028ba917c3633ac3ed312836f9d4f8aeb5356c76/packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts#L318

File: packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts [hunk 2]
```diff
@@ -273,15 +273,7 @@ export const getStyles = (props: IPanelStyleProps): IPanelStyles => {
       !isOpen && isAnimating && isOnRightSide && AnimationClassNames.slideRightOut40,
       focusTrapZoneClassName
     ],
-    commands: [
-      classNames.commands,
-      {
-        marginTop: 18
```
Comment (nkanaki) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: I think the marginTop is still needed. That shouldn't be an issue, right @phkuo?

Comment (nkanaki) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: perhaps same with marginLeft as well

File: packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts [hunk 3]
```diff
@@ -273,34 +273,18 @@ export const getStyles = (props: IPanelStyleProps): IPanelStyles => {
...
+    commands: [classNames.commands],
     navigation: [
       classNames.navigation,
       {
         display: 'flex',
-        justifyContent: 'flex-end'
+        justifyContent: 'space-between'
       },
       hasCustomNavigation && {
         height: commandBarHeight
```
Comment (nkanaki) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: I would double check this, but changing justifyContent: 'space-between' might be breaking customNavigation which actually needs flex-end

File: packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts (hunk 2 above)
Comment (phkuo) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: If it's on a flex parent and not a flex child, then yes, margins are fine.

File: packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts (hunk 3 above)
Comment (phkuo) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: What does custom nav look like? flex-end is going to cause the title to right-align. You can try putting flex-grow:1 on the title to make sure it stays left aligned, since then it'll fill up all that left-hand side space and the text inside will left-align to that. Just have to make sure the flex parent there is always 100% width, no more and no less.

Comment (aneeshack4) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: Why does customNavigation need flex-end?

Comment (nkanaki) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: @aneeshack4 @phkuo it doesn't! just checked

File: packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts (hunk 2 above)
Comment (aneeshack4) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts]: Cool that should fix the screener regression

**Comment by phkuo** [comment]
https://css-tricks.com/the-peculiar-magic-of-flexbox-and-auto-margins/ interesting read
